### PR TITLE
Refactor(web-react): Use `@local/tests` path alias in tests

### PR DESCRIPTION
**Pull request by literat**
## Description

1. Introducing barrel files for `tests` directory
2. Introducing configuration for `@local/tests` path alias - tsconfig + ESlint
3. Use path alias in all tests
4. Refactoring `useIcon` mock because of path changes and crazy Jest and JavaScript mock hoisting
5. Fixing @pavelklibani 's typo in `data-testId` -> `data-testid`

### Additional context

I suggest doing a review per commit. The settings themselves are in different commits than changes in test files.

### Issue reference

https://jira.almacareer.tech/browse/DS-1611
https://jira.almacareer.tech/browse/DS-1612

### CODE REVIEW

File: packages/web-react/src/components/Accordion/__tests__/Accordion.test.tsx [hunk 1]
```diff
@@ -1,9 +1,7 @@
 import '@testing-library/jest-dom';
 import { render } from '@testing-library/react';
 import React from 'react';
-import { classNamePrefixProviderTest } from '../../../../tests/providerTests/classNamePrefixProviderTest';
-import { restPropsTest } from '../../../../tests/providerTests/restPropsTest';
-import { stylePropsTest } from '../../../../tests/providerTests/stylePropsTest';
+import { classNamePrefixProviderTest, restPropsTest, stylePropsTest } from '@local/tests';
```
Comment (literat) [inline packages/web-react/src/components/Accordion/__tests__/Accordion.test.tsx:4]: **todo (if-minor):** We need to setup the correct import order. Maybe @curdaj can help with it.

File: packages/web-react/jest.config.ts [hunk 2]
```diff
@@ -13,6 +13,13 @@ const config = {
   // Automatically clear mock calls, instances, contexts and results before every test.
   // https://jestjs.io/docs/configuration#clearmocks-boolean
   clearMocks: true,
+
+  // A map from regular expressions to module names or to arrays of module names
+  // https://jestjs.io/docs/configuration#modulenamemapper-objectstring-string--arraystring
+  moduleNameMapper: {
```
Comment (literat) [inline packages/web-react/jest.config.ts:19]: **suggestion (if-minor):** Consider using `ts-jest` and https://kulshekhar.github.io/ts-jest/docs/getting-started/paths-mapping/

File: packages/web-react/tsconfig.json [hunk 3]
```diff
@@ -25,7 +25,11 @@
     "esModuleInterop": true,
     "skipLibCheck": true,
     "lib": ["es2016", "dom", "dom.iterable"],
-    "types": ["jest", "node"]
+    "types": ["jest", "node"],
+    "paths": {
+      "@local/*": ["./*"],
```
Comment (literat) [inline packages/web-react/tsconfig.json:30]: **suggestion (if-minor):** 
```suggestion
      "@local/tests": ["./tests/index.ts"]
```

Comment (literat) [reply packages/web-react/tsconfig.json:30]: We do not want to allow `@local/src/**`.

Comment (literat) [reply packages/web-react/tsconfig.json:30]: Fixed.

**Comment by crishpeen** [comment]
FYI, I created two issues for this. As we decided on Long Topics [DS-1611](https://jira.almacareer.tech/browse/DS-1611), [DS-1612](https://jira.almacareer.tech/browse/DS-1612).

**Review by pavelklibani — APPROVED** [review]

File: packages/web-react/.eslintrc.js [hunk 4]
```diff
@@ -73,9 +88,14 @@ module.exports = {
       {
         groups: ['builtin', 'external', 'internal', 'parent', 'sibling', 'index'],
         pathGroups: [
+          // {
```
[the remark this answers is not in the document]
Comment (literat) [reply packages/web-react/.eslintrc.js]: Line removed.

**Pull request merged**